core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -385,6 +368,141 @@ sealed abstract class MultiSigSigner
+          val updatedWit = uwtx.updateWitness(inputIndex.toInt, p2wpkhWit)
+
+          Future.successful(
+            P2SHTxSigComponent(updatedWit, inputIndex, output, flags))
+        case (nestedSpendingInfo: P2SHNestedSegwitV0UTXOSpendingInfo,
+              _: P2WSHWitnessSPKV0) =>
+          val outpoint = spendingInfo.outPoint
+          val redeemScript = nestedSpendingInfo.redeemScript
+          val oldInput = unsignedTx.inputs(inputIndex.toInt)
+
+          val input =
+            TransactionInput(outpoint, EmptyScriptSignature, oldInput.sequence)
+
+          val updatedTx =
+            unsignedTx.updateInput(inputIndex.toInt, input)
+
+          val updatedOutput =
+            TransactionOutput(spendingInfo.amount, redeemScript)
+
+          val updatedUTXOInfo = BitcoinUTXOSpendingInfo(
+            outpoint,
+            updatedOutput,
+            spendingInfo.signers,
+            None,
+            spendingInfo.scriptWitnessOpt,
+            spendingInfo.hashType,
+            spendingInfo.conditionalPath)
+
+          val signedTxEither =
+            BitcoinSigner
+              .sign(updatedUTXOInfo, updatedTx, isDummySignature)
+              .map(_.transaction)
+
+          signedTxEither.map { signedTx =>
+            val i = signedTx.inputs(inputIndex.toInt)
+
+            val p2sh =
+              P2SHScriptSignature(i.scriptSignature, redeemScript)
+
+            val signedInput =
+              TransactionInput(i.previousOutput, p2sh, i.sequence)
+
+            val signedInputs =
+              signedTx.inputs.updated(inputIndex.toInt, signedInput)
+
+            val finalTx = signedTx match {
+              case btx: BaseTransaction =>
+                BaseTransaction(btx.version,
+                                signedInputs,
+                                btx.outputs,
+                                btx.lockTime)
+              case wtx: WitnessTransaction =>
+                WitnessTransaction(wtx.version,
+                                   signedInputs,
+                                   wtx.outputs,
+                                   wtx.lockTime,
+                                   wtx.witness)
+            }
+            P2SHTxSigComponent(finalTx, inputIndex, output, flags)
+          }
+        case (_, _) =>
+          val signedSigComponentF = BitcoinSigner.sign(
+            spendingInfoToSatisfy.nestedSpendingInfo,
+            unsignedTx,
+            isDummySignature,
+            spendingInfoToSatisfy.nestedSpendingInfo)
+
+          val scriptSigF = signedSigComponentF.map { signedSigComponent =>
+            P2SHScriptSignature(signedSigComponent.scriptSignature,
+                                spendingInfoToSatisfy.redeemScript)
+          }
+
+          updateScriptSigInSigComponent(unsignedTx,
+                                        inputIndex.toInt,
+                                        output,
+                                        scriptSigF)
+      }
```

Comment:
Is there a reason this case differs from the one above it? Looking at the old implementation in `TxBuilder` above it looks like they used to do the same thing